### PR TITLE
🎉 New Source: Databricks

### PR DESCRIPTION
**Pull request by ameyabapat-bsft**
## What
Source Databricks - Jdbc
This PR looks to solve https://github.com/airbytehq/airbyte/issues/28161

## How
Java Jdbc Source Connector for Databricks on top of JdbcSourceConnector.

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Comment by ameyabapat-bsft** [comment]
airbyte-ci connectors --name=source-databricks test

**Comment by marcosmarxm** [comment]
@ameyabapat-bsft please move the PR to Ready to Review whatever is done. Please ask my review then

**Comment by itaseskii** [comment]
@marcosmarxm I'll take over this :)

**Comment by ameyabapat-bsft** [comment]
Added output of `./gradlew :airbyte-integrations:connectors:source-databricks:integrationTest`
[Source Databricks Integration Tests v0.txt](https://github.com/airbytehq/airbyte/files/12096027/Source.Databricks.Integration.Tests.v0.txt)

**Comment by James-Liebler** [comment]
Hello, just curious as I have also been working on something similar, does this have a side effect of taking about 5/6 minutes for initial connection to databricks?

**Comment by ameyabapat-bsft** [comment]
> Hello, just curious as I have also been working on something similar, does this have a side effect of taking about 5/6 minutes for initial connection to databricks?

Hi @James-Liebler Do you mean the time that databricks takes to start/launch the warehouse? I see the delay of ~3mins for the first connection if a warehouse(cloud resource ie: EC2 machine) is not running. The subsequent queries run at normal speed depending on the size of warehouse and cluster without extra delay.

**Comment by James-Liebler** [comment]
> 

Yes that's exactly what I'm seeing, for our set up it takes about 5 minutes, once the warehouse has been spun up then it will connect almost immediately.

Thank you for confirmation! What steps are remaining for this to be merged?

**Comment by ameyabapat-bsft** [comment]
> > 
> 
> Yes that's exactly what I'm seeing, for our set up it takes about 5 minutes, once the warehouse has been spun up then it will connect almost immediately.
> 
> Thank you for confirmation! What steps are remaining for this to be merged?

There is a acceptance test failing which could be due to recent changes in platform. Once it is fixed and tests start to pass, I will resolve review comments if any. @itaseskii is reviewing it.

**Comment by ameyabapat-bsft** [comment]
@marcosmarxm could you provide an update on fixes in acceptance test? 
This PR is blocked for more than 3 weeks.Thanks 
cc: @itaseskii

File: airbyte-integrations/connectors/source-databricks/acceptance-test-config.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# See [Connector Acceptance Tests](https://docs.airbyte.com/connector-development/testing-connectors/connector-acceptance-tests-reference)
+# for more information about how to configure these tests
+connector_image: airbyte/source-databricks:dev
+acceptance_tests:
+  spec:
+    tests:
+      - spec_path: "src/test-integration/resources/expected_spec.json"
+        config_path: "src/test-integration/resources/dummy_config.json"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-databricks/acceptance-test-config.yml:8]: @ameyabapat-bsft @itaseskii did you try to comment this part?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-databricks/acceptance-test-config.yml (hunk 1 above)
Comment (ameyabapat-bsft) [reply airbyte-integrations/connectors/source-databricks/acceptance-test-config.yml:8]: commenting this part throws error
```
  source-databricks git:(source-databrick-v0) ✗ sh acceptance-test-docker.sh
[+] Building 0.3s (10/10) FINISHED                                                                                                                                                                 docker:desktop-linux
 => [internal] load build definition from Dockerfile                                                                                                                                                               0.0s
 => => transferring dockerfile: 670B                                                                                                                                                                               0.0s
 => [internal] load .dockerignore                                                                                                                                                                                  0.0s
 => => transferring context: 2B                                                                                                                                                                                    0.0s
 => [internal] load metadata for docker.io/airbyte/integration-base-java:dev                                                                                                                                       0.0s
 => [internal] load build context                                                                                                                                                                                  0.0s
 => => transferring context: 137B                                                                                                                                                                                  0.0s
 => [build 1/4] FROM docker.io/airbyte/integration-base-java:dev                                                                                                                                                   0.0s
 => CACHED [build 2/4] WORKDIR /airbyte                                                                                                                                                                            0.0s
 => CACHED [build 3/4] COPY build/distributions/source-databricks*.tar source-databricks.tar                                                                                                                       0.0s
 => CACHED [build 4/4] RUN tar xf source-databricks.tar --strip-components=1 && rm -rf source-databricks.tar                                                                                                       0.0s
 => CACHED [stage-1 3/3] COPY --from=build /airbyte /airbyte                                                                                                                                                       0.0s
 => exporting to image                                                                                                                                                                                             0.1s
 => => exporting layers                                                                                                                                                                                            0.0s
 => => writing image sha256:23fd53a0d35fc012d5849686fe9906b88c37c3a7c34b7e64e5554033f7a7ff56                                                                                                                       0.1s
 => => naming to docker.io/airbyte/source-databricks:dev                                                                                                                                                           0.0s

What's Next?
  View summary of image vulnerabilities and recommendations → docker scout quickview
latest: Pulling from airbyte/connector-acceptance-test
Digest: sha256:273f283618e92e4af6b167d5ba64c6e88cf389d199462995bff9fef64e4404a1
Status: Image is up to date for airbyte/connector-acceptance-test:latest
docker.io/airbyte/connector-acceptance-test:latest

What's Next?
  View summary of image vulnerabilities and recommendations → docker scout quickview airbyte/connector-acceptance-test:latest
============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-6.2.5, py-1.11.0, pluggy-1.2.0
rootdir: /test_input
plugins: requests-mock-1.9.3, cov-3.0.0, mock-3.6.1, timeout-1.4.2, sugar-0.9.7, anyio-3.7.1, hypothesis-6.82.3
collected 0 items / 6 errors
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/_pytest/main.py", line 269, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/_pytest/main.py", line 322, in _main
INTERNALERROR>     config.hook.pytest_collection(session=session)
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/pluggy/_hooks.py", line 433, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/pluggy/_manager.py", line 112, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/pluggy/_callers.py", line 155, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/pluggy/_result.py", line 108, in get_result
INTERNALERROR>     raise exc.with_traceback(exc.__traceback__)
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/pluggy/_callers.py", line 80, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/_pytest/main.py", line 333, in pytest_collection
INTERNALERROR>     session.perform_collect()
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/_pytest/main.py", line 637, in perform_collect
INTERNALERROR>     hook.pytest_collection_modifyitems(
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/pluggy/_hooks.py", line 433, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/pluggy/_manager.py", line 112, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/pluggy/_callers.py", line 155, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/pluggy/_result.py", line 108, in get_result
INTERNALERROR>     raise exc.with_traceback(exc.__traceback__)
INTERNALERROR>   File "/usr/local/lib/python3.10/site-packages/pluggy/_callers.py", line 80, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/app/connector_acceptance_test/plugin.py", line 120, in pytest_collection_modifyitems
INTERNALERROR>     config = load_config(config.getoption("--acceptance-test-config"))
INTERNALERROR>   File "/app/connector_acceptance_test/utils/common.py", line 38, in load_config
INTERNALERROR>     return Config.parse_obj(data)
INTERNALERROR>   File "pydantic/main.py", line 521, in pydantic.main.BaseModel.parse_obj
INTERNALERROR>   File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
INTERNALERROR> pydantic.error_wrappers.ValidationError: 1 validation error for Config
INTERNALERROR> acceptance_tests
INTERNALERROR>   none is not an allowed value (type=type_error.none.not_allowed)


======================== 2 warnings, 6 errors in 1.77s
```

**Comment by ameyabapat-bsft** [comment]
@marcosmarxm I have messaged the config/secrets to run test. Were you able to run them? were you able to reproduce the issue? Let me know if you need any help.

**Comment by marcosmarxm** [comment]
> @marcosmarxm I have messaged the config/secrets to run test. Were you able to run them? were you able to reproduce the issue? Let me know if you need any help.

As already said our team is dedicated to review August Hackathon contributions and if there are some space we're going to try to help other contributions.

**Comment by ameyabapat-bsft** [comment]
Hello @marcosmarxm, Hope you are doing well.
Can you check the issues in acceptance tests? I have slack ref https://airbytehq.slack.com/archives/C027KKE4BCZ/p1692365566365049?thread_ts=1692167732.700609&cid=C027KKE4BCZ 
My PR is blocked for many days. It would be great help if you could take a look at it. 
cc: @itaseskii

**Comment by ameyabapat-bsft** [comment]
Attaching Passed Build and Tests:
[gradlew build with test-17-10-23.txt](https://github.com/airbytehq/airbyte/files/12929770/gradlew.build.with.test-17-10-23.txt)
[integration_tests-17-10-23.txt](https://github.com/airbytehq/airbyte/files/12929775/integration_tests-17-10-23.txt)
@itaseskii @marcosmarxm Could we initiate the review process? Let me know if you need anything.

**Comment by marcosmarxm** [comment]
Hello! Thanks for your contribution. We’ve noticed it’s been a while since you received an update. After internal discussions, the team has decided not to accept new contributions for Java sources and destinations at the moment. Airbyte is currently revamping the shared core destinations codebase and plans to release a new Java destinations CDK in 2024.
The team is working on improving general usability, speed through asynchronous loading, and Typing and Deduplication (Destinations V2). Accepting contributions in the current state has led to confusion among users. In the meantime, feel free to build and host the images independently. Share them with users interested in your connector until the official Java CDK is released and ready to accept new connectors. Thanks for your understanding and feel free to reach me directly in Slack to discuss the topic if you needed.

**Comment by ameyabapat-bsft** [comment]
Hi @CPAPI-104 , 
You can create custom databricks source connector image from this branch and add to your workspace. 
Ex: ` docker build ./airbyte-integrations/connectors/source-databricks -t airbyte/source-databricks:v0`
we have been using it ourselves with airbyte 0.50.34  for our use cases. It is working well without any performance limits. 
Also, @marcosmarxm It seems there is a need for databricks source connector, are we now accepting new db sources? If yes, I will create a new PR from my older one and will try to get it published. 
Many Thanks !!

**Comment by CPAPI-104** [comment]
@ameyabapat-bsft We're using airbyte cloud so I dont think that solution will work for us. Anything else you'd recommend?

**Comment by rafael-frazao-sa** [comment]
Hey @ameyabapat-bsft that connector is only working in older versions right? If you for some reason udpated your Airbyte instance, can you share a bit about the changes inside that connector?

I see that some image, entrypoints and params doesnt exist anymore in recent Airbyte versions.

I would be glad to hear your experience maintaining that custom connector.

Thank you. :)

**Pull request closed without merging**